src/Pdf.php

```diff
@@ -168,8 +204,14 @@ public function getImageData($pathToImage)
 
         $imagick->readImage(sprintf('%s[%s]', $this->pdfFile, $this->page - 1));
 
+        $imagick->setImageFormat($this->determineOutputFormat($pathToImage));
+
+        $imagick->transformImageColorspace($this->outputColorSpace);
+
         $imagick->setFormat($this->determineOutputFormat($pathToImage));
 
+        file_put_contents($pathToImage, $imagick);
```

Comment:
I have issues with this line, see this previous PR https://github.com/spatie/pdf-to-image/pull/4 for more details.

This makes assumptions about how people are storing files.